log/syslog/syslog.go

```diff
@@ -0,0 +1,46 @@
+// Copyright 2020 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+package syslog
+
+import (
+	"log/syslog"
+
+	"github.com/sirupsen/logrus"
+	logrus_syslog "github.com/sirupsen/logrus/hooks/syslog"
+)
+
+type SyslogHook struct {
+	loglevel logrus.Level
+	*logrus_syslog.SyslogHook
+}
```

Comment:
@kacf This is all it takes (: 
Go isn't that bad :1st_place_medal:

Reply:
You're right, it works "by accident", because we are lucky (or because the guy knows this limitation), and [`Fire()` doesn't call `Levels()` anywhere](https://github.com/sirupsen/logrus/blob/master/hooks/syslog/syslog.go#L28-L51). If it had, it would not have worked. :-)

But this means you have already fixed it then, and my previous comment no longer applies?

Just to be clear, we still log everything at one level inside the syslog, is that right? This only controls which ones of the Go-side levels get logged at all?

Reply:
> You're right, it works "by accident", because we are lucky (or because the guy knows this limitation), and [`Fire()` doesn't call `Levels()` anywhere](https://github.com/sirupsen/logrus/blob/master/hooks/syslog/syslog.go#L28-L51). If it had, it would not have worked. :-)
> 

As far as I can tell, the `Levels` call are only used when the hook is loaded. So the levels is sort of: `This hook will subscribe to these log-levels. Then when that log-level is called, the logger goes through a map with the current log-level called. If this map has a hook pointer, then that hook is fired. Therefore, with this change, the syslog should not be called for levels below what is set by the user on the CLI now.`

> But this means you have already fixed it then, and my previous comment no longer applies?
> 
This should fix it yes :)

> Just to be clear, we still log everything at one level inside the syslog, is that right? This only controls which ones of the Go-side levels get logged at all?

It seems to me from the `Fire()` function in the syslog-hook that it will actually log at the appropriate levels. So the initializer level is basically a no-op. Except for the `LOG_USER` which we now added.

Reply:
:+1: